google/cloud/testing_util/validate_metadata.cc

```diff
@@ -45,42 +51,23 @@ StatusOr<std::map<std::string, std::string> > ExtractMDFromHeader(
     std::smatch match_res;
     std::string s = i->str();
     bool const matched = std::regex_match(s, match_res, assign_re);
-    if (!matched) {
-      return Status(
-          StatusCode::kInvalidArgument,
-          "Bad header format. The header should be a series of \"a=b\" "
-          "delimited with \"&\", but is \"" +
-              s + "\"");
-    }
+    EXPECT_TRUE(matched)
+        << "Bad header format. The header should be a series of \"a=b\" "
+           "delimited with \"&\", but is \"" +
+               s + "\"";
+    if (!matched) continue;
     bool const inserted =
         res.insert(std::make_pair(match_res[1].str(), match_res[2].str()))
             .second;
-    if (!inserted) {
-      return Status(
-          StatusCode::kInvalidArgument,
-          "Param " + match_res[1].str() + " is listed more then once");
-    }
+    EXPECT_TRUE(inserted) << "Param " + match_res[1].str() +
+                                 " is listed more than once";
   }
   return res;
 }
 
-StatusOr<std::map<std::string, std::string> > ExtractMDFromHeaders(
-    std::multimap<std::string, std::string> const& headers) {
-  auto param_header = headers.equal_range("x-goog-request-params");
-  if (param_header.first == param_header.second) {
-    return Status(StatusCode::kInvalidArgument, "Expected header not found");
-  }
-  if (std::distance(param_header.first, param_header.second) > 1U) {
-    return Status(StatusCode::kInvalidArgument, "Multiple headers found");
-  }
-  return ExtractMDFromHeader(param_header.first->second);
-}
-
-/// A poor man's check if a value matches a glob used in URL patterns.
-bool ValueMatchesPattern(std::string const& val, std::string const& pattern) {
-  std::string regexified_pattern =
-      regex_replace(pattern, std::regex("\\*"), std::string("[^/]+"));
-  return std::regex_match(val, std::regex(regexified_pattern));
+MATCHER_P(ContainsStdRegex, pattern, "ContainsRegex using std::regex") {
```

Comment:
Where does `ContainsRegex` not meet the mark?

Reply:
So, it seems then that the Windows googletest was built with `GTEST_USES_SIMPLE_RE`, and everything else with `GTEST_USES_POSIX_RE`.  If we can't change that, then perhaps a comment to that effect would avoid confusion.

Reply:
> perhaps a comment to that effect would avoid confusion.

Good idea. I added one in: https://github.com/googleapis/google-cloud-cpp/pull/9383/commits/d0ab8ea53a37f39a163425b37abfe532804d2e59